modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/rest/DDMRESTDataProvider.java

```diff
@@ -373,10 +373,18 @@ private DDMDataProviderResponse _getData(
 				).build(),
 				false);
 
-		String response = jsonWebServiceClient.doGet(
-			absoluteURL, _getParametersArray(allParameters));
+		String response = null;
 
-		jsonWebServiceClient.destroy();
+		try {
+			response = jsonWebServiceClient.doGet(
+				absoluteURL, _getParametersArray(allParameters));
+		}
+		catch (JSONWebServiceException jsonWebServiceException) {
+			throw jsonWebServiceException;
+		}
```

Comment:
Hi @jcampoy, do we really need this `catch`?  Wouldn't `try/finally` be enough?

Reply:
we have to throw the exception, see the caller that is catching and handling them...

Reply:
if the try-finally throws up the exception, in that case I am right with the change :D

Reply:
The `try` block will throw the exception anyway, but the `catch` would be necessary in this case if we need to handle the exception some other way

Reply:
@carolmariaabb I am going to resend the PR with your suggestions! thanks!

Reply:
@carolmariaabb the changes have been done